PocketKit/Sources/PocketKit/Home/Views/Cards/CardView.swift

```diff
@@ -149,6 +151,20 @@ private extension CardView {
         }
     }
 
+    /// We are legally allowed to open the item in reader view except one of the following conditions is met:
+    /// a) the item is not an article (i.e. it was not parseable)
+    /// b) the item is an image
+    /// c) the item is a video
+    /// - Returns: true if we can open this item in the reader
+    func isReaderSupported() -> Bool {
```

Comment:
added this method that checks if the underlying `Item` supports reader. This is done because while we observe `SavedItem` via `@Query` (which allows to dynamically update the view when an item is saved, archived or favorited), swift observation does not necessarily guarantees that changes to related items will be immediately reflected (thus for example the `isArticle` property might not be immediately be updated if a user saves a recommendation and open it right after: this could cause the item to be still opened in a web view instead of the reader).
We could also do a `@Query` to observe both the `SavedItem` and the underlying `Item` but that would create unnecessary overhead, since we only need this when a user navigates to a saved item from the related recommendation.